packages/compass-components/src/hooks/use-toast.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { createContext, useCallback, useContext, useState } from 'react';
+import type { ToastVariant } from '..';
+import { css } from '..';
+import { Toast } from '..';
+
+type ToastProperties = {
+  title?: React.ReactNode;
+  body: React.ReactNode;
+  variant: ToastVariant;
+  progress?: number;
+  timeout?: number;
+};
+
+type ToastState = ToastProperties & {
+  timeoutRef?: ReturnType<typeof setTimeout>;
+};
+
+type ToastId = string;
+
+type ToastContextState = {
+  toasts: Record<ToastId, ToastState>;
+  setToasts: (toasts: Record<ToastId, ToastState>) => void;
+};
+
+interface ToastActions {
+  openToast: (id: ToastId, toastProperties: ToastProperties) => void;
+  closeToast: (id: ToastId) => void;
+}
+
+const ToastContext = createContext<ToastContextState>({
+  toasts: {},
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  setToasts: (t: Record<ToastId, ToastProperties>) => {
+    return;
+  },
+});
+
+const toastStyles = css({
+  button: {
+    position: 'absolute',
+  },
+});
+
+const ToastPile = (): React.ReactElement => {
+  const { toasts } = useContext(ToastContext);
+  const { closeToast } = useGlobalToast();
+
+  return (
+    <>
+      {Object.entries(toasts).map(
+        ([id, { title, body, variant, progress }]) => (
+          <Toast
+            className={toastStyles}
+            key={id}
+            title={title}
+            body={body}
+            variant={variant}
+            progress={progress}
+            open={true}
+            close={() => closeToast(id)}
+          />
```

Comment:
Seems like by default toast component renders itself with position fixed which means that multiple toasts are just stacked on top of each other and you can't see them:

This is how three toasts look in UI:

<img width="446" alt="image" src="https://user-images.githubusercontent.com/5036933/152946289-f4a65581-0cb1-4ab7-9c2e-f50c7e47a5d9.png">

And this is the DOM (just to prove that there are multiple of them):

<img width="611" alt="image" src="https://user-images.githubusercontent.com/5036933/152946412-947a12cb-ee81-47d6-a0f4-b9a8977cb25e.png">

I think we either need to ask leafygreen team to implement toast stack or work around that by having our own fixed container here + adding some styles to the toast to override its position (we probably would need to do both because I don't see leafygreen implementing this change any time soom)

Reply:
> fixed container here + adding some styles

That's probably easy enough and probably we could rely on the fact that `position: fixed` should always be on the Toast element itself.

We decided not to make this change here, but worth a try

Reply:
Ah yeah, I see now that they render them in a portal (on top of having fixed position, why?) and in this case don't allow to override the portal node (or disable the portal similar to other components). I will ask them how are we even supposed to handle this. If you want to merge this as-is that's fine, but basically with how leafygreen works right now we can't make multiple toasts work on the same screen and so maybe we shouldn't even "allow" this on the hook level, because we are basically allowing broken behavior that way

Reply:
At hook level is partially handled, I think is best if LG fixes it, but since you have to pass an id, creating multiple toasts has to be intentional. Ie. the `copy-connection-string` will not create multiple toasts, but the next one will replace the previous one if is there.

Reply:
Yeah, we definitely handle it on our side! What I mean is what happens when I need to add a toast with another id somewhere else in the app? It will result in a broken UI with overlapping toasts, do we just agree that this is okay for now and wait for leafygreen team to provide the solution in the future?